### PR TITLE
Replace panics triggered by malformed requests by graceful handling.

### LINKED ISSUE #228: Invalid queries can cause the node tasks to panic
**Bug Description**

If a query is made with an invalid input (concretely a string containing a null character) the task that handles it panics.
This is fortunately not very severe since tokio gracefully handles the panic and remains alive, responding to further queries.
But it is nevertheless an issue that should be fixed and this input problem should be handled correctly.

**Steps to Reproduce**

Make a query which expects a string as input (e.g., any consensus query that requires a block hash) with a malformed block hash with a 0 byte in the string.

**Expected Result**

Invalid request or similar error is returned.

**Actual Result**

The task handling the request panics and the connection is terminated.

**Versions**
- Software Version: All existing versions.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Fix #228 

## Changes

- Handle invalid input by responding with InvalidArgument status code.
- Remove needless copies of byte buffers when passing them to Haskell. This is not directly related to the issue above, but it is in the same place in the code and so I noticed it.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 - The macOS installer no longer overwrites the service files when reinstalling.
 - Cache smart contract modules on startup from existing state to improve smart
   contract execution.
+- Make consensus queries more robust, by validating input more extensively.
```
Comment (EmilLa1) [inline CHANGELOG.md:11]: Maybe add the GRPC code that invalid ones will return i.e., `InvalidArgument`?

**Review by EmilLa1 — COMMENTED** [review]
Should we validate more? Etc. the length of the block hashes submitted?

**Comment by abizjak** [comment]
> Should we validate more? Etc. the length of the block hashes submitted?

This is validated by Haskell functions, e.g., `decodeBlockHash` fails if the hash is incorrect length.

We could change this setup so that we would not be allocating CStrings and instead validated everything Rust-side and just passed a *const u8 to Haskell, which could assume that the data is valid.

But that is a bigger refactoring, this is just a bugfix.

**Review by EmilLa1 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (abizjak) [reply CHANGELOG.md:11]: Done.

**Review by td202 — APPROVED** [review]
One suggestion (see comment). Otherwise seems fine.

File: concordium-node/src/consensus_ffi/ffi.rs [hunk 2]
```diff
@@ -571,12 +561,12 @@ impl ConsensusContainer {
...
     }
 
-    pub fn get_block_info(&self, block_hash: &str) -> String {
-        let c_str = CString::new(block_hash).unwrap();
-        wrap_c_call_string!(self, consensus, |consensus| getBlockInfo(
+    pub fn get_block_info(&self, block_hash: &str) -> anyhow::Result<String> {
+        let c_str = CString::new(block_hash)?;
+        Ok(wrap_c_call_string!(self, consensus, |consensus| getBlockInfo(
             consensus,
             c_str.as_ptr() as *const u8
```
[the remark this answers is not in the document]
Comment (abizjak) [reply concordium-node/src/consensus_ffi/ffi.rs]: Changed.

**Pull request merged**